homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -62,10 +71,16 @@
     BINARY_SENSOR_CROSSLINE_DETECTED_POLLED: _CROSSLINE_DETECTED_PARAMS,
     BINARY_SENSOR_ONLINE: ("Online", DEVICE_CLASS_CONNECTIVITY, None),
 }
-BINARY_SENSORS = {
-    k: dict(zip((SENSOR_NAME, SENSOR_DEVICE_CLASS, SENSOR_EVENT_CODE), v))
-    for k, v in RAW_BINARY_SENSORS.items()
-}
+BINARY_SENSORS: tuple[AmcrestSensorEntityDescription, ...] = tuple(
+    AmcrestSensorEntityDescription(
+        key=key,
+        device_class=device_class,
+        name=name,
+        event_code=event_code,
+    )
+    for key, (name, device_class, event_code) in RAW_BINARY_SENSORS.items()
```

Comment:
Just a thought, should the whole `RAW_BINARY_SENSOR` just be replaced by a single tuple that has the entity descriptions? I think that would make it all more readable

Reply:
That sounds good, there is a bit of verbosity that the raw binary sensors helps with, but making this flat should be much more readable.